PDDL.py

```diff
@@ -95,9 +97,29 @@ def parse_predicates(self, group):
                 arguments[untyped_variables.pop(0)] = 'object'
             self.predicates[predicate_name] = arguments
 
-    #-----------------------------------------------
+    # -----------------------------------------------
+    # Parse types
+    # -----------------------------------------------
+    def parse_types(self, types):
+        untyped_names = []
+        while types:
+            t = types.pop(0)
+            if self.types.get(t):
+                raise Exception('Type ' + t + ' redefined')
```

Comment:
Some domains contain such redefinitions, some of them are harder to catch due to cyclical constructions:
```
t1 t3 - t2
t2 - t1
```
Perhaps ``'Redefined supertype of ' + t`` is more clear.